opentelemetry/src/global/internal_logging.rs

```diff
@@ -104,11 +123,46 @@ macro_rules! otel_error {
         {
             tracing::error!(name: $name, target: env!("CARGO_PKG_NAME"), "");
         }
+        #[cfg(not(feature = "internal-logs"))]
+        {
+            #[allow(unused_variables)]
+            {
+
+            }
+        }
     };
     (name: $name:expr, $($key:ident = $value:expr),+ $(,)?) => {
         #[cfg(feature = "internal-logs")]
         {
-            tracing::error!(name: $name, target: env!("CARGO_PKG_NAME"), $($key = $value),+, "");
+            tracing::error!(name: $name,
+                            target: env!("CARGO_PKG_NAME"),
+                            $($key = {
+                                    $crate::format_value!($value)
+                            }),+,
+                            ""
+                    )
+        }
+        #[cfg(not(feature = "internal-logs"))]
+        {
+            {
+                let _ = ($name, $($value),+);
+
+            }
         }
     };
 }
+
+/// Helper macro to format a value using Debug if available, falling back to Display
+#[macro_export]
+macro_rules! format_value {
+    ($value:expr) => {{
+        // Try Debug first
+        let debug_result = std::fmt::format(format_args!("{:?}", $value));
+        if debug_result.starts_with('<') || debug_result.contains("::") {
```

Comment:
Can you add some details on why is this needed? Not sure I am able to understand this part...

Reply:
This was to try formatting with `Debug`, and fallback to `Display` if there is any error. But now in latest commit I have simplified now to always use `Debug`.